_posts/2018-11-09-simple-slackbot.md

```diff
@@ -0,0 +1,67 @@
+---
+layout: post
+title:  "Simple Slackbot"
+date: 2018-11-09 00:03:30 +0200
+categories: javascript
+author_name: Sergii Paryzhskyi
+author_url : /author/sergii_paryzhskyi
+author_avatar: sergii_paryzhskyi
+read_time : 7
+show_related_posts: false
+square_related: recommend-wolf
+feature_image: posts/2018-11-09-simple-slackbot/poster.jpg
+---
+
+Recently I've built a small slackbot that is posting with certain regularity in a specified Slack channel. I will tell you about the idea behind it and also share some technical details.
+
+
+## The Idea
+
+The idea to build such a bot came from one colleague of mine - [Jacek Smolak][jacek]. At that point he was reading the book ["97 Things Every Programmer Should Know"][book-link]. In order to share those programmer tips with others and also remind them to himself, he came up with idea of posting each of them one by one on a daily basis in appropriate Slack channel. For us this Slack channel is #craft, you probably have something similar amoung your channels in Slack.
+
+
+## Implementation
+
+It is dead simple. As it was written in JavaScript, the easiest way was to use the official [slack npm package][slack-npm]. Posting something in specific channel is just a one-liner that returns a Promise that we can handle depending if it is success or failure:
+
+<em class="snippet-description">Post message to Slack</em>
+```js
+slackClient.chat.postMessage({
+    channel: '#craft',
+    message: '...'
+});
+```
+
+The very same interface is used for sending direct messages, so without changing anything you can specify your own Slack-name instead of channel. In this case you will be getting the messages from the bot directly. Its quite handy for testing purposes as well.
+Let's have a look on how we are going to pick the quotes that should be posted. The simpliest source would be a json file, that can look like this:
```

Comment:
How about a new paragraph here? You even make it a new line, but the rendering will just make it one by one.

Reply:
yep 👍